src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -130,4 +133,48 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
       case None => List.empty
     }
   }
+
+  def getEntitySubtrees(workspaceNamespace: String, workspaceName: String, entityType: String, entityNames: Seq[String], txn: RawlsTransaction): Seq[Entity] = txn withGraph { db =>
+    def nameFilter = new PipeFunction[Vertex, java.lang.Boolean] {
+      override def compute(v: Vertex) = entityNames.contains(v.getProperty("_name"))
+    }
+
+    def isLeaf = new PipeFunction[LoopPipe.LoopBundle[Vertex], java.lang.Boolean] {
+      override def compute(bundle: LoopPipe.LoopBundle[Vertex]): java.lang.Boolean = { true }
+    }
+
+    def emitAll = new PipeFunction[LoopPipe.LoopBundle[Vertex], java.lang.Boolean] {
+      override def compute(bundle: LoopPipe.LoopBundle[Vertex]): java.lang.Boolean = { true }
+    }
+
+    val subtreeEntities = getWorkspaceVertex(db, workspaceNamespace, workspaceName) match {
+      case Some(w) => {
+        val topLevelEntities = new GremlinPipeline(w).out(entityType).filter(nameFilter).iterator().map(loadEntity(_, workspaceNamespace, workspaceName)).toList
+        val nextLevelEntities = new GremlinPipeline(w).out(entityType).filter(nameFilter).as("outLoop").out().dedup().loop("outLoop", isLeaf, emitAll).iterator().map(loadEntity(_, workspaceNamespace, workspaceName)).toList
```

Comment:
The names are misleading- you're right. isLeaf can be removed and emitAll can be called twice in the loop pipe. As for being able to emit the topLevelEntities without a second pipe, you'd think that it would be possible but there was no apparent way to keep them with the nextLevelEntities pipeline. I tried all of the various pipes that sounded like they'd do that and none seemed to retain them properly.